src/acp/policy.ts

```diff
@@ -0,0 +1,179 @@
+ * ```
+ *
+ * Mark a given [[SolidDataset]] as containing [[Policy]]'s, and save it to the given URL.
+ *
+ * @param url URL to save this Access Policy SolidDataset at.
+ * @param dataset The SolidDataset containing Access Policies to save.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ */
+export async function savePolicyDatasetAt(
+  url: Url | UrlString,
+  dataset: PolicyDataset,
+  options: Partial<
+    typeof internal_defaultFetchOptions
+  > = internal_defaultFetchOptions
+): ReturnType<typeof saveSolidDatasetAt> {
+  url = internal_toIriString(url);
+  let datasetThing = getThing(dataset, url) ?? createThing({ url: url });
+  datasetThing = setUrl(datasetThing, rdf.type, acp.AccessPolicyResource);
+  dataset = setThing(dataset, datasetThing);
+
+  return saveSolidDatasetAt(url, dataset, options);
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new, empty [[Policy]].
+ *
+ * @param url URL that identifies this Access Policy.
+ */
+export function createPolicy(url: Url | UrlString): Policy {
+  url = internal_toIriString(url);
+  let policyThing = createThing({ url: url });
+  policyThing = setUrl(policyThing, rdf.type, acp.AccessPolicy);
+
+  return policyThing;
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Get the [[Policy]] with the given URL from an [[PolicyDataset]].
+ *
+ * @param policyResource The Resource that contains the given Access Policy.
+ * @param url URL that identifies this Access Policy.
+ * @returns The requested Access Policy, if it exists, or `null` if it does not.
+ */
+export function getPolicy(
+  policyResource: PolicyDataset,
+  url: Url | UrlString
+): Policy | null {
+  const foundThing = getThing(policyResource, url);
+  if (
+    foundThing === null ||
+    getUrl(foundThing, rdf.type) !== acp.AccessPolicy
+  ) {
+    return null;
+  }
+
+  return foundThing;
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Get all [[Policy]]'s in a given [[PolicyDataset]].
+ *
+ * @param policyResource The Resource that contains Access Policies.
+ */
+export function getPolicyAll(policyResource: PolicyDataset): Policy[] {
+  const foundThings = getThingAll(policyResource);
+  const foundPolicies = foundThings.filter(
+    (thing) =>
+      !isThingLocal(thing) && getUrl(thing, rdf.type) === acp.AccessPolicy
```

Comment:
There is a possibility that an ACP might have several types, and that `getUrl` does not return the one we expect here. I'd suggest using `getUrlAll(thing, rdf.type).includes(acp.AccessPolicy)`.

Reply:
Good catch: 2ba0d5b26cbd94bc2a6e5036baf9fb0234e6f015

Reply:
Yep - as with my previous comment, I'd actually just include an RDF-type-aware convenience function - something like 'isThingOfType(thing, type)` or 'hasThingType(thing, type)`.